tests/baselines/reference/checkJsxChildrenProperty13.errors.txt

```diff
@@ -1,33 +0,0 @@
-tests/cases/conformance/jsx/file.tsx(12,30): error TS2710: 'children' are specified twice. The attribute named 'children' will be overwritten.
-
-
-==== tests/cases/conformance/jsx/file.tsx (1 errors) ====
-    import React = require('react');
-    
-    interface ButtonProp {
-        a: number,
-        b: string,
-        children: Button;
-    }
-    
-    class Button extends React.Component<ButtonProp, any> {
-        render() {
-            // Error children are specified twice
-            return (<InnerButton {...this.props} children="hi">
-                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-!!! error TS2710: 'children' are specified twice. The attribute named 'children' will be overwritten.

```

Comment:
It’s unclear what this test was supposed to be testing. `children` are actually specified three times: first in the spread, then in the named attribute, then in the JSX children. IMO, it should never be an error to overwrite a spread property with a named declaration (in JSX props _or_ object literals), but it would be quite reasonable to issue an error when a) a named declaration overwrites a named declaration, or b) a spread property overrides a named declaration (which is what this PR is about). And I’m counting JSX children (like the tag contents) as _named_, even though the name is implicit.

By way of example, here are the tests I think should probably exist at some point:

```ts
interface AnchorProps { href: string; children: string }
declare const props: AnchorProps;

const a1 = <a {...props} href="#" />; // ok: analgous to `p = { ...props, href: '#' }`
const a2 = <a {...props} children="Link" />; // ok: same as above
const a3 = <a {...props}>Link</a>; // ok: same as above

const a4 = <a href="#" {...props} />; // not ok (as of this PR): analgous to `p = { href: '#', ...props }`
const a5 = <a children="Link" {...props} />; // not ok: same as above

const a6 = <a children="Link">Link</a>; // I guess not ok, but kinda linty and unrelated to this PR
```

Reply:
```
const a6 = <a children="Link">Link</a>; // I guess not ok, but kinda linty and unrelated to this PR
```
is the case the old error was mostly concerned with catching~

Reply:
Yeah, I just reverted the JSX changes since it's just going to hold up this PR to try to unify what appears to be an actually-quite-different-and-complex check.